src/coreclr/System.Private.CoreLib/src/System/Reflection/RuntimeMethodInfo.CoreCLR.cs

```diff
@@ -312,7 +312,7 @@ public override MethodImplAttributes GetMethodImplementationFlags()
             {
                 StackAllocedArguments argStorage = default;
                 Span<object?> copyOfParameters = new(ref argStorage._arg0, 1);
-                ReadOnlySpan<object?> parameters = new(in parameter);
+                object?[] parameters = new object?[] { parameter };
```

Comment:
This allocation will regress this special code path for single argument.

I think we should keep the `ReadOnlySpan<object>` argument and keep this as `ReadOnlySpan<object>`.

Instead we should add an additional `object[]` argument to `CheckArguments` method. This additional `object[]` is only going to be used for the write-back and the write-back will be only done when the argument is non-null.

This place will pass in this extra argument as null.

Reply:
Search for `// Copy modified values out. This should be done only with ByRef or Type.Missing parameters.` for examples on how copy-back is done elsewhere.

Reply:
We can use the existing copy-back mechanism instead of adding an additional argument - see the recent commit.